src/views/NutritionView.vue

```diff
@@ -308,60 +308,31 @@ export default {
-    //       else if(keyToMatch.match(/sugar.*/) && "sugar" in this.nutritionData[0].data.nutrients){
-    //         this.currentProductData[nutrimentId].data = this.nutritionData[0].data.nutrients.sugar[0].value;
-    //       }
-    //       else if(keyToMatch.match(/carbohydrate.*/) && "carbohydrate" in this.nutritionData[0].data.nutrients){
-    //         this.currentProductData[nutrimentId].data = this.nutritionData[0].data.nutrients.carbohydrate[0].value;
-    //      }
-    //       else if(keyToMatch.match(/protein.*/) && "protein" in this.nutritionData[0].data.nutrients){
-    //         this.currentProductData[nutrimentId].data = this.nutritionData[0].data.nutrients.protein[0].value;
-    //       }
-    //       else if(keyToMatch.match(/salt.*/) && "salt" in this.nutritionData[0].data.nutrients){
-    //         this.currentProductData[nutrimentId].data = this.nutritionData[0].data.nutrients.salt[0].value
-    //       }
-    //       else if(keyToMatch.match(/fiber.*/) && "fiber" in this.nutritionData[0].data.nutrients){
-    //         this.currentProductData[nutrimentId].data = this.nutritionData[0].data.nutrients.fiber[0].value;
-    //       }
-    //     }// if
-    //   }// for
-    // },    
+    },   
 
     setNutritionData: function(){
+      const namingCorrection = {sugar: 'sugars', protein: 'proteins', carbohydrate: 'carbohydrates'};
+      const correctNaming = (robotoffNaming) => `nutriment_${namingCorrection[robotoffNaming] || robotoffNaming}`
+
       for (const nutrimentId of Object.keys(this.nutritionData[0].data.nutrients)) {
         let nutrimentKey = nutrimentId.replace(/_/g, '-');
-        // two letter nutrimentId's are separated by underscore ("_") in robotoff
-        // while in nutritions.json they are separated with hypen ("-")
-        nutrimentKey = "nutriment_"+nutrimentKey;
-        if(nutrimentKey in this.currentProductData)
-        {
-          console.log(`${nutrimentKey} exists`);
-          // when key exists update the key with nutrition value
-        }
-        else{
-          console.log(`${nutrimentKey} does not exits`);
-          // when key does not exists in this.currentProductData, add the new key
+        let correctName = correctNaming(nutrimentKey);
 
+        if(nutrimentId=="energy"){
+          correctName = `nutriment_energy-${this.nutritionData[0].data.nutrients[nutrimentId][0].unit}`;
         }
-        // this.currentProductData[nutrimentKey].data = this.nutritionData[0].data.nutrients[nutrimentId][0].value;     
+
+        if(!(correctName in this.currentProductData))
+        {
+          const data = {
+            id: nutrimentId,
+            data: "",
+            unit: this.nutritionData[0].data.nutrients[nutrimentId][0].unit,
+            visible: true,
+          };
+          this.currentProductData[correctName] = data;
+        }// if
+        this.currentProductData[correctName].data = this.nutritionData[0].data.nutrients[nutrimentId][0].value;     
```

Comment:
Here you set some data and you erase it just after

Reply:
> Here you set some data and you erase it just after

You mean hard-code a key name to test?